xapian-core/cluster/cluster.cc

```diff
@@ -110,18 +121,16 @@ TermListGroup::add_document(const Document &document) {
 }
 
 void
-TermListGroup::add_documents(DocumentSource &docs) {
+TermListGroup::add_documents(MSetDocumentSource docs) {
```

Comment:
Problem here is, that the parameter 'MSetDocumentSource docs' is modified inside add_documents( .. ), so I can't cast it as a const because that leads to an error. And I can't even pass this by reference because I do not want this object to change its state outside this function. That's why I pass it by value.
It won't make a very huge performance difference because these functions are more like a pre-processing thing.